src/node_crypto.cc

```diff
@@ -4320,7 +4360,7 @@ SignBase::Error Verify::VerifyFinal(const char* key_pem,
   if (!initialised_)
     return kSignNotInitialised;
 
-  ClearErrorOnReturn clear_error_on_return;
+  // ClearErrorOnReturn clear_error_on_return;
 
```

Comment:
You make a compelling case, but I would have to look more closely and don't have time right now. I think there was a bug recently where sign failure was causing subsequent operations to fail because stack wasn't cleared, but that just means its important to do, not that its important to do here, so this needs careful thought.